### DISCUSSION TITLE
Support java game

### DISCUSSION THREAD

**Discussion by KriTeiler** [OnionUI/Onion#831] [Ideas]
I would love to see Java emulation. Mobile games are also a kind of classic.

**Comment by schmurtzm** [comment]
I don't think that it is technically possible sadly.

**Reply by Starbrush** [reply]
J2ME can't run on miyoo unfortunately, so it's impossible

**Reply by Mark-B97** [reply]
What about RA core SquirrelJ2ME?

**Reply by XK9274** [reply]
> J2ME can't run on miyoo unfortunately, so it's impossible

It can, i've had them working but it's clunky, very slow, bad input and the JDK is about 300mb.

Worms crazy golf java game w/ Free2JME
![image](https://github.com/OnionUI/Onion/assets/47260768/0be45921-a989-49ff-8242-a58cd6d280f8)
![image](https://github.com/OnionUI/Onion/assets/47260768/5946d31c-d6ce-45f0-8909-a8b7d15914a6)


SquirrelJ2ME currently doesn't do much at all, it's still in development and only runs a single test rom:

![image](https://github.com/OnionUI/Onion/assets/47260768/2126780b-6ae7-42b4-a79f-9829bf5b3543)

You can view the timeline here: https://multiphasicapps.net/timeline

**Reply by Starbrush** [reply]
Sometimes I forget how close the community is, amazing work XK! It's great to know it's in development

**Reply by iAbsolute** [reply]
Sorry for the question, but how can mmp run psx but not java? (I'm a bit of a layman, that's why I asked)? I wanted to play Diamond Rush on mmp

**Reply by fengfrw** [reply]
https://github.com/aweigit/freej2me-miyoomini

**Comment by ruaxbk** [comment]
J2ME emulator based on freej2me, supporting 3D games
reference
[freej2me](https://github.com/hex007/freej2me)
[J2ME-Loader](https://github.com/nikita36078/J2ME-Loader)
[JL-Mod](https://github.com/woesss/JL-Mod)

Tested on miyoomini, gkdminiplus, rg28xx, ubuntu18

compile:
The front-end uses sdl2, needs to compile sdl_interface

The sound uses sdl2-mixer, needs to compile libaudio.so

3D game support, M3G needs to compile libm3g.so (depending on EGL, GLES1), Mascot Capsule v3 needs to compile libmicro3d.so (depending on EGL, GLES2)

run:
export JAVA_TOOL_OPTIONS='-Xverify:none -Djava.util.prefs.systemRoot=./.java -Djava.util.prefs.userRoot=./.java/.userPrefs -Djava.library.path=./'

java -jar freej2me-sdl.jar /home/game.jar 240 320 100
240 320 refers to the game resolution, and 100 refers to the volume (0-100)
